### ISSUE TITLE
Model alone makes different predictions compared to trainer + model

### ISSUE THREAD

**Issue by JanRuettinger** [PyTorchLightning/pytorch-lightning#2772]
## 🐛 Bug

I have trained a model and stored it in a checkpoint file. When I load the model from the checkpoint file and make predictions for my test set I get different results when I use the raw model compared to when I use the trainer class + model.

### Problematic code snippets

Using the trainer class
```
trainer = Trainer(gpus=1)
ckpt_path='/content/gdrive/My Drive/colab_projects/colab_demo/checkpoints/model.ckpt'
model = TransferLearningModel.load_from_checkpoint(str(ckpt_path), **config)
trainer.test(model)
```

Not using the trainer class
```
model = TransferLearningModel.load_from_checkpoint(str(ckpt_path), **config)
model.freeze()
model.cuda()
with torch.no_grad():
  for x,y in dataloader:
    x = x.cuda()
    output = model(x)
    ....

```

### To Reproduce
I have setup a colab notebook which reproduces the bug.

https://drive.google.com/file/d/16Vjcx_-NZ-pBA2mZJRSfytGTmDC8P9xY/view?usp=sharing

Happy to add more information when needed.

**Comment by Borda** [comment]
it is interesting, that shall not happen, are you using the same data without shuffle?
@SkafteNicki mind have a look at it? :]

**Comment by rohitgr7** [comment]
By results, if you are referring to the DataFrames you printed in the notebook, then I think the results looks the same to me.

**Comment by raynardj** [comment]
Could it be not running ```model = model.eval()``` when not using trainer class.

```eval()``` does more than freeze, it uses overall mean and std for batchnorm instead of running mean and std. Also it will turn off the dropout.

**Comment by rohitgr7** [comment]
@raynardj `model.freeze()` does that automatically I think.

**Comment by raynardj** [comment]
> @raynardj `model.freeze()` does that automatically I think.

u r right, eval() is there.

I also took a close look at the comparison of 2 dataframe and they r the same

**Comment by agemagician** [comment]
@rohitgr7 @raynardj @Borda @SkafteNicki Any update on this issue?

I am facing a similar behavior in my prediction.

**Comment by rohitgr7** [comment]
@agemagician Can you share code sample we can look at?

**Comment by agemagician** [comment]
Sure, I will write a Colab notebook for my code and share it, but did you figure out the problem from the above-shared notebook?

**Comment by rohitgr7** [comment]
Not yet. We still don't find the results that are referred in the notebook by author. If the results are the dataframes then both are same. There is no difference in final prediction.

**Comment by Borda** [comment]
> @agemagician Can you share code sample we can look at?

this is the notebook above - https://colab.research.google.com/drive/16Vjcx_-NZ-pBA2mZJRSfytGTmDC8P9xY

**Comment by agemagician** [comment]
After implementing the code again in Colab, I figured out the problem.
In my case, the input pipeline for test and prediction was different and that's why I didn't get the same results.

**Comment by JanRuettinger** [comment]
@raynardj the results I get are not the same as you can see in the two pictures below. They are similar but not the same. Do you have any pointers where to look for errors?

<img width="503" alt="Screenshot 2020-08-06 at 19 40 39" src="https://user-images.githubusercontent.com/8582703/89564120-d015e080-d81c-11ea-8ebf-01d9b07ced09.png">
<img width="516" alt="Screenshot 2020-08-06 at 19 40 32" src="https://user-images.githubusercontent.com/8582703/89564128-d441fe00-d81c-11ea-8b81-9143f32564e0.png">

**Comment by skhan61** [comment]
Hello,

I've trained a model in PyTorch Lightning and saved it to a checkpoint. When I load this model from the checkpoint and use it for predictions directly, I get different results than when I use the Trainer class to handle the predictions. I've ensured the model is in evaluation mode (model.eval()), using the same input data. Could you help me understand what might be causing these differences?

Thank you!

**Comment by zoezhou1999** [comment]
Hi @skhan61 
I met the same issue. Did you solve it?
Thank you!

Best
Yuhongze
